### PR TITLE
CPLP-1791-getapplication

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1791

### CODE REVIEW

File: src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs [hunk 1]
```diff
@@ -528,7 +528,7 @@ public async Task<int> SetInvitationStatusAsync(string iamUserId)
 
     public async Task<RegistrationData> GetRegistrationDataAsync(Guid applicationId, string iamUserId)
     {
-        var registrationData = await _portalRepositories.GetInstance<IUserRepository>().GetRegistrationDataUntrackedAsync(applicationId, iamUserId).ConfigureAwait(false);
+        var registrationData = await _portalRepositories.GetInstance<IUserRepository>().GetRegistrationDataUntrackedAsync(applicationId, iamUserId, _settings.DocumentTypeIds).ConfigureAwait(false);
```
Comment (Phil91) [inline src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:531]: Are the documentTypeIds from the settings used for another use case as well?

Comment (qxz2mqe) [reply src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:531]: yes documenttypeid is being used for another use case

File: src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ public UserRepository(PortalDbContext portalDbContext)
             })
             .AsAsyncEnumerable();
 
-    public Task<RegistrationData?> GetRegistrationDataUntrackedAsync(Guid applicationId, string iamUserId) =>
+    public Task<RegistrationData?> GetRegistrationDataUntrackedAsync(Guid applicationId, string iamUserId, IEnumerable<DocumentTypeId> documentTypes) =>
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs:57]: Could you please provide unit tests for the repository

File: src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs [hunk 3]
```diff
@@ -65,7 +65,7 @@ public UserRepository(PortalDbContext portalDbContext)
                 company!.Id,
                 company.Name,
                 company.CompanyAssignedRoles!.Select(companyAssignedRole => companyAssignedRole.CompanyRoleId),
-                company.CompanyUsers.SelectMany(companyUser => companyUser!.Documents!.Select(document => new RegistrationDocumentNames(document.DocumentName))),
+                company.CompanyUsers.SelectMany(companyUser => companyUser!.Documents!.Where(document=>documentTypes.Contains(document.DocumentTypeId)).Select(document => new RegistrationDocumentNames(document.DocumentName))),
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs:68]: Could you please provide unit tests

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs:57]: Added Unit Test

**Review by Phil91 — APPROVED** [review]

**Pull request merged**